### PR TITLE
[TIMOB-23527] iOS10: Support the UserNotifications framework

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23527

### CODE REVIEW

**Comment by vijaysingh-axway** [comment]
@hansemannn, It's good if we can create module for same. Because -
1. As we are trying to minimise Core,  so creating module will fulfil this requirement.
2. Maintenance will be easy. 
3. Any unseen issue which may arise due to these changes, will not happen in sdk.
4. And your PR of ticket TIMOB-24266 is already merged, so it will be easy to use sdk delegates in module.

**Comment by hansemannn** [comment]
I would agree, but then we:
- miss a core feature in the core sdk
- will have issues maining parity across iOS versions (which is highly critical)
- May experience duplicate events and handlers, since complex use cases like silent pushes also require other core features like background-services that only can be completed once.

What I would suggest is that we create a Singleton NotificationManager class in the sdk that is clearly separated into both frameworks, but shares a common api endpoint.

Let me still think about the module approach, but sooner or later, it will need to come to the SDK anyway. If it will be a customer wanting to have parity or the move to a higher minimum SDK version.

File: iphone/Classes/TiAppiOSProxy.m [hunk 1]
```diff
@@ -507,15 +495,50 @@ - (NSArray *)supportedUserActivityTypes
...
-    returnVal = [[self formatUserNotificationSettings:notificationSettings] retain];
-  },
-      YES);
+  DEPRECATED_REPLACED(@"App.iOS.currentUserNotificationSettings", @"7.2.0", @"App.iOS.NotificationCenter.requestUserNotificationSettings");
 
-  return [returnVal autorelease];
-  ;
+  if ([TiUtils isIOS10OrGreater]) {
+    DebugLog(@"[ERROR] Please use Ti.App.iOS.NotificationCenter.requestUserNotificationSettings in iOS 10 and later to request user notification settings asynchronously.");
+    return;
```
Comment (vijaysingh-axway) [inline iphone/Classes/TiAppiOSProxy.m]: If this function is previously getting used, now it will not return relevant data in iOS >= 10.

**Review by vijaysingh-axway — CHANGES_REQUESTED** [review]
TiAppiOSNotificationCategoryProxy.m, TiAppiOSNotificationActionProxy.m, TiAppiOSNotificationActionProxy.h files are deleted. Are these files getting used internally only? What if developer is using?

File: apidoc/Titanium/App/iOS/NotificationCenter.yml [hunk 2]
```diff
@@ -0,0 +1,216 @@
+---
+name: Titanium.App.iOS.NotificationCenter
+summary: |
+    The top-level App iOS Notification Center module, available only to iOS devices running iOS 8 or later. It is
+    used to control scheduled notifications and receive details about the system-wide notification settings.
+
+extends: Titanium.Module
+platforms: [iphone, ipad]
+osver: {ios: {min: "8.0"}}
+since: "7.2.0"
```
[the remark this answers is not in the document]
Comment (hansemannn) [reply apidoc/Titanium/App/iOS/NotificationCenter.yml]: Updated!

File: iphone/Classes/TiAppiOSProxy.m (hunk 1 above)
Comment (hansemannn) [reply iphone/Classes/TiAppiOSProxy.m]: Updated!

File: iphone/Classes/TiAppiOSLocalNotificationProxy.h [hunk 3]
```diff
@@ -1,21 +1,23 @@
...
   @private
-  UILocalNotification *_notification;
+  id _notification;
 }
 
-@property (nonatomic, retain) UILocalNotification *notification;
+@property (nonatomic, retain) id notification;
 
-- (void)cancel:(id)args;
+- (void)cancel:(id)used;
```
[the remark this answers is not in the document]
Comment (hansemannn) [reply iphone/Classes/TiAppiOSLocalNotificationProxy.h]: Updated!

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Comment by vijaysingh-axway** [comment]
@hansemannn Can you resolve conflict please. Thanks!

**Pull request merged**